### ISSUE TITLE
Esquema de la base de datos

### ISSUE THREAD

**Issue by islobell** [islobell/CAIN#2]
El esquema de la base de datos es el siguiente:

![musicdb](https://cloud.githubusercontent.com/assets/9166208/6284644/b0520bc4-b8f2-11e4-96b0-892e0456cc20.png)

Donde:
- Tabla **Tracks**, tiene los siguientes campos:
  - ID: clave principal.
  - MBID: identificador unívoco de la pista de audio.
  - Title: título de la pista de audio.
  - Year: año de publicación.
  - Duration: duración en milisegundos.  
  - PathDst: ubicación del archivo una vez catalogado.
  - Status: estado de la canción; habrán 3 estados: 1) sin resultados; 2) catalogada; 3) no catalogada.
- Tabla **Tags**, tiene los siguientes campos:
  - ID: clave principal.
  - TrackID: clave foránea referida a la tabla 'Tracks'.
  - Name:  nombre de la etiqueta.
  - Content: contenido de la etiqueta.
  - Description: descripción para hacerse una idea de a qué se refiere.
- Tabla **TrackTags**, tiene los siguientes campos:
  - TrackID: clave foránea referida a la tabla 'Tracks'.
  - TragID: clave foránea referida a la tabla 'Trags'.
- Tabla **Artists**, tiene los siguientes campos:
  - ID: clave principal.
  - MBID: identificador unívoco del artista.
  - Name: nombre del artista.
- Tabla **TrackArtists**, tiene los siguientes campos:
  - TrackID: clave foránea referida a la tabla 'Tracks'.
  - ArtistID: clave foránea referida a la tabla 'Artists'.
- Tabla **Albums**, tiene los siguientes campos:
  - ID: clave principal.
  - MBID: identificador unívoco del álbum.
  - Title: título del álbum.
  - Year: año de publicación.
- Tabla **TrackAlbums**, tiene los siguientes campos:
  - TrackID: clave foránea referida a la tabla 'Tracks'.
  - AlbumID: clave foránea referida a la tabla 'Albums'.

**Comment by aoltra** [comment]
En el track podrías incluir el año. Es una dato común a todas las pista y puede ser interesante a la hora de realizar futuras consultas como por ejemplo _música de los 80_

**Comment by islobell** [comment]
Correcto. El año es una información valiosa para catalogar la música, pero sería una etiqueta que asociaríamos a una o varias canciones y por lo tanto debería ir en la tabla 'Tags'. ¿Estás de acuerdo?

**Comment by aoltra** [comment]
No lo veo así :|. Una etiqueta es algo que puede estar o no asociada a una canción. Por ejemplo, _voz femenina_. Algunas canciones la tendrán, otras no. O _baile_. Sin embargo, el año estará en todas, es algo así como el bitrate.. todas lo tendrán. ¿No?

**Comment by islobell** [comment]
Una etiqueta siempre estará asociada al menos a una canción, aunque esa etiqueta puede que no se asocie a todas las canciones. Una etiqueta por sí sola no tiene sentido, sino que hace referencia a un objeto (en este caso una o varias canciones). 

El año no es algo intrínseco al fichero de audio como el bitrate, el samplerate o el códec de audio, que siempre deben estar presentes pues describen las propiedades del audio que contiene el archivo. En cambio, el año (o cualquier otra etiqueta), si se encuentra presente, formará parte de los tags del fichero de audio. De hecho un archivo de audio puede no soportar tags o no tener.

Por este motivo, en la base de datos he separado estos 2 conjuntos de informaciones: las propiedades intrínsecas del archivo de audio van en la tabla 'Tracks'; mientras que los tags que pueda contener, la información obtenida al consultar enciclopedias musicales como MusicBrainz y la información aportada por el usuario van en la tabla 'Tags'.

**Comment by aoltra** [comment]
Umm!! Creo que ya se donde está nuestra diferencia de ideas. Es que para mi el concepto no es fichero (o track), es canción. Realmente información como el bitrate o los canales no tiene demasiada importancia ya que no afecta a la canción (si que lo hace al fichero), de hecho varios de esos atributos los eliminaría (no me van a hacer falta en un futuro)

Entiendo lo que me comentas sobre diferenciar entre información intrínseca y la que no lo es. pero  por ejemplo el año puede estar incluido dentro de los tags del mp3 ...  O, visto por el otro lado, el album y el artista muy posiblemente lo localicemos de Music Brainz y si embargo lo colocamos en tablas, no en etiquetas.  

La verdad  es que el tema de decidir que es etiqueta o no es interesante :) . Tu planteamiento lógico es una posibilidad, el mio es un planteamiento menos lógico y más funcional: para que tener entradas en una tabla diferente y crear una relación cuando _año de grabación_ es una característica  propia de la canción?

Y por cierto, ¿y el nombre de la canción?  ;)

**Comment by islobell** [comment]
Ok. Ya está cambiado.
